Unifies and refactors documentation (#83)
JuliaNLSolvers/Manifolds.jl: docs/src/interface.md

[Pull Request](https://github.com/JuliaNLSolvers/Manifolds.jl/compare) to add it here.

```@autodocs
Modules = [Manifolds, ManifoldsBase]
Pages = ["ManifoldsBase.jl"]
Order = [:type, :function]
```

> As a first manifold, a reduced simplified version of [`Euclidean`](@ref) the interface provides

Reviewer: ```suggestion
`DefaultManifold` is a simplified version of [`Euclidean`](@ref) and demonstrates a basic interface implementation.
```
